src/js/ripple/amount.js

```diff
@@ -325,7 +327,9 @@ Amount.prototype.invert = function() {
 
 Amount.prototype.canonicalize = function(roundingMode) {
   if (this._is_native) {
-    this._value = this._value.round(6, BigNumber.ROUND_DOWN);
+    var value = this._value.round(6, BigNumber.ROUND_DOWN);
+    var isOverflow = value.abs().greaterThan(Amount.bi_xrp_max);
+    this._value = isOverflow ? new BigNumber(NaN) : value;
```

Comment:
I agree, these should be consistent. The question is whether it is better to throw or use NaN. This problem applies throughout the Amount class; some methods throw, others return NaN. Throwing is nice because the client code finds out about the issue right away.

The counter-argument might be as follows. Let's say ripple-trade adds two amounts using the Amount class to display the sum on the page and they forget to check for exceptions. If we use NaN, then presumably the box on the page would show "NaN", whereas if we throw, it could break the whole page. Arguably, breaking the whole page is ok because technically it is a bug that the client is not checking for errors and it should be fixed immediately.